cmd/api/src/queries/graph.go

```diff
@@ -523,6 +520,24 @@ func (s *GraphQuery) RawCypherSearch(ctx context.Context, pQuery PreparedQuery,
 	return graphResponse, nil
 }
 
+func applyTimeoutReduction(queryWeight int64, availableRuntime time.Duration) time.Duration {
+	availableRuntimeInt := int64(availableRuntime.Seconds())
+	// The weight of the query is divided by 5 to get a runtime reduction factor, in a way that:
+	// weights of 4 or less get the full runtime duration
+	// weights of 5-9 will get 1/2 the runtime duration
+	// weights of 10-15 will get 1/3 the runtime duration
+	// and so on until the max weight of 50 gets 1/11 the runtime duration
+
+	// NOTE reductionFactor will be the math.Floor() of the result of the division below
+	reductionFactor := 1 + (queryWeight / 5)
+	availableRuntimeInt /= reductionFactor
+	availableRuntime = time.Duration(availableRuntimeInt) * time.Second
+
+	log.Infof("Cypher query cost is: %d. Reduction factor for query is: %d. Available timeout for query is now set to: %d seconds", queryWeight, reductionFactor, availableRuntime)
```

Comment:
Not ideal as it would require another parameter for just a log, but I wonder if we'd get a lot more out of this log if we included the stripped query to easily correlate

perhaps including it in the log event somehow 🤔

Reply:
Actually yeah, we probably do need the query here. Correlation is otherwise virtually impossible